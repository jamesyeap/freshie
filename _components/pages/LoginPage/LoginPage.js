import React from 'react';
import styled from 'styled-components';
import { Container } from '../../_atoms/Container';
import { BrandHeaderText } from '../../_atoms/Text';
import { Header } from '../../_molecules/Header';
import { TextInput } from '../../_molecules/TextInput';
import { BigButton, TextButton, Checkbox } from '../../_atoms/Button';
import { Formik } from 'formik';
import * as Yup from 'yup';
import { loginAsync_API } from '../../../_utilities/_api/Auth';

const OptionsContainer = styled.View`
	flexDirection: row;
	justifyContent: space-between;
	alignItems: center;
	width: 300px;
	margin: 12px;
`;

const LoginSchema = Yup.object().shape({
	username: Yup.string()
		   .required('Email is required!'),
	password: Yup.string()
		     .required('Please enter a password!')

})

export default function LoginPage(props) {

	const handleLogin = (values) => {
		console.log(values);
		loginAsync_API(values);
	}

	return (
		<Formik
		 initialValues={{ username: '', password: '' }}
		 validationSchema={LoginSchema}
		 onSubmit={values => handleLogin(values)}
		>
		{({ handleChange, handleBlur, handleSubmit, values, touched, errors }) => (
		<Container>
			<BrandHeaderText>freshie</BrandHeaderText>
			<Header
			 headerText="Hello, Welcome Back"
			 subheaderText="Enter your credentials to continue"
			/>

			<TextInput
			label="Username"
			placeholder="ahBengWithBang"
			onChangeText={handleChange('username')}
			onBlur={handleBlur('username')}
			value={values.username}
			feedbackMessage={errors.username}
			touched={touched.username}
			autoCapitalize="none"
			/>

			<TextInput
			label="Password"
			onChangeText={handleChange('password')}
			onBlur={handleBlur('password')}
			value={values.password}
			feedbackMessage={errors.password}
			touched={touched.password}
			autoCapitalize="none"
			secureTextEntry
			/>

			<OptionsContainer>
				<Checkbox label="Remember me"/>
				<TextButton label="Forgot Password?" />
			</OptionsContainer>

			{/* <BigButton label="Sign In" state="active" onPress={() => props.navigation.push("Client")}/> */}
			<BigButton label="Sign In" state="active" onPress={handleSubmit}/>

			<TextButton label="Don't have an account?" onPress={() => props.navigation.push("Signup")} buttonStyle={{ marginTop: 20 }}/>
		</Container>
		)}
		</Formik>
	)
}