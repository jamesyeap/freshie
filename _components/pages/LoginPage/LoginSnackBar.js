import * as React from 'react';
import { View, StyleSheet } from 'react-native';
import { Button, Snackbar } from 'react-native-paper';
import { ActivityIndicator } from 'react-native-paper';
import { SemiBoldText } from '../../_atoms/Text';
import { connect } from 'react-redux';

function mapStateToProps(state) {
	const { loading, error } = state.auth;
	return { loading, error };
}

export function LoginSnackBar (props) {
  const [visible, setVisible] = React.useState(true);

  const onToggleSnackBar = () => setVisible(!visible);

  const onDismissSnackBar = () => setVisible(false);

  return (
      <Snackbar
        visible={props.visible}
	duration={100000}
        onDismiss={onDismissSnackBar}
	wrapperStyle={{
		marginBottom: 30,
		alignItems: "center",
		width: "100%",
	}}
	style={{
		flexDirection: 'row',
		alignItems: 'center',
		width: "90%",
		
		justifyContent: "space-between",
		borderColor: "black",
	}}
	>
	Logging you in
      </Snackbar>
  );
};

export default connect(mapStateToProps)(LoginSnackBar);

/*
return (
	<Snackbar
	  visible={visible}
	  duration={100000}
	  onDismiss={onDismissSnackBar}
	  action={{
	    label: 'Undo',
	    onPress: () => {
	      // Do something
	    },
	  }}
	  wrapperStyle={{
		  marginBottom: 30,
		  alignItems: "center"
	  }}
	  style={{
		  width: "90%",
		  backgroundColor: "red"
	  }}
	  >
	  Hey there! I'm a Snackbar.
	</Snackbar>
    );
  };
  */