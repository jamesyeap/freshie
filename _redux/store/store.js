import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import RootReducer from '../reducers/index';

export const store = createStore(RootReducer, applyMiddleware(thunk, logger));