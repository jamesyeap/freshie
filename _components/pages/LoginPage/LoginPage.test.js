import React from 'react'
import { Provider } from 'react-redux';
import { store } from '../../../_redux/store/store'
import { render, screen, fireEvent } from '@testing-library/react-native';
import 'jest-styled-components'

import LoginPage from './LoginPage'

const setup = () => {
	const utils = render(
		<Provider store={store}>
			<LoginPage />
		</Provider>
	)

	return utils;
}

test('user can key in username and password', () => {
	const { getByLabelText, getByTestId } = render(
		<Provider store={store}>
			<LoginPage />
		</Provider>
	)

	const usernameInput = getByTestId('username-input')
	const fakeUsernameText = 'mark'
	fireEvent.changeText(usernameInput, fakeUsernameText)
	console.log(usernameInput.value)

})

