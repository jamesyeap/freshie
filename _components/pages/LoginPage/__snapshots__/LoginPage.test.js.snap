// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-start",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Inter_500Medium",
          "fontSize": 20,
          "lineHeight": 28,
        },
      ]
    }
  >
    freshie
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
          "marginBottom": 31,
          "marginTop": 77,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Inter_500Medium",
            "fontSize": 24,
            "lineHeight": 32,
          },
        ]
      }
    >
      Hello, Welcome Back
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#989898",
            "fontFamily": "Inter_500Medium",
            "fontSize": 14,
            "lineHeight": 20,
          },
        ]
      }
    >
      Enter your credentials to continue
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
          "justifyContent": "center",
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2D3748",
            "fontFamily": "Inter_500Medium",
            "fontSize": 16,
            "lineHeight": 24,
            "marginBottom": 8,
          },
        ]
      }
    >
      Username
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      label="Username"
      minHeight={40}
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="enter your username!"
      rejectResponderTermination={true}
      style={
        Array [
          Object {},
          Object {
            "borderColor": "#E2E8F0",
            "borderRadius": 6,
            "borderWidth": 1,
            "fontFamily": "Inter_400Regular",
            "fontSize": 16,
            "lineHeight": 24,
            "paddingBottom": 5,
            "paddingLeft": 5,
            "textAlign": "left",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
      width={320}
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
          "justifyContent": "center",
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2D3748",
            "fontFamily": "Inter_500Medium",
            "fontSize": 16,
            "lineHeight": 24,
            "marginBottom": 8,
          },
        ]
      }
    >
      Password
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      label="Password"
      minHeight={40}
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="enter your password!"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Array [
          Object {},
          Object {
            "borderColor": "#E2E8F0",
            "borderRadius": 6,
            "borderWidth": 1,
            "fontFamily": "Inter_400Regular",
            "fontSize": 16,
            "lineHeight": 24,
            "paddingBottom": 5,
            "paddingLeft": 5,
            "textAlign": "left",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
      width={320}
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 12,
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 12,
          "width": 300,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessibilityLiveRegion="polite"
        accessibilityRole="checkbox"
        accessibilityState={
          Object {
            "checked": false,
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 18,
              "height": 36,
              "padding": 6,
              "width": 36,
            },
          ]
        }
        testID="checkbox"
      >
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <Text />
          <View
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "borderColor": "rgba(0, 0, 0, 0.54)",
                  "borderWidth": 0,
                  "height": 14,
                  "width": 14,
                }
              }
            />
          </View>
        </View>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Inter_600SemiBold",
              "fontSize": 14,
              "lineHeight": 20,
              "marginRight": 9,
            },
          ]
        }
      >
        Remember me
      </Text>
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Inter_600SemiBold",
              "fontSize": 14,
              "lineHeight": 20,
            },
          ]
        }
      >
        Forgot Password?
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#285E61",
        "borderRadius": 6,
        "flexDirection": "column",
        "height": 48,
        "justifyContent": "center",
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 0,
        "opacity": 1,
        "width": 300,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Inter_600SemiBold",
            "fontSize": 18,
            "lineHeight": 28,
          },
        ]
      }
    >
      Sign In
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
        "marginBottom": 0,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 20,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Inter_600SemiBold",
            "fontSize": 14,
            "lineHeight": 20,
          },
        ]
      }
    >
      Don't have an account?
    </Text>
  </View>
</RCTSafeAreaView>
`;
